src/components/UserProfile/UserProfile.jsx

```diff
@@ -1,13 +1,30 @@
-import './UserProfile.css'
+import { useEffect } from "react";
+import "./UserProfile.css";
+import { useState } from "react";
+
+const UserProfile = () => {
+  const [users, setUsers] = useState([]);
+
+  useEffect(() => {
+    fetch("https://jsonplaceholder.typicode.com/users")
+      .then((res) => res.json())
+      .then((data) => setUsers(data));
+  }, []);
+
+  if (users.length === 0) {
+    return <div>Loading...</div>;
```

Comment:
Excellent! adding a Loading indicator is always a good practice, however the logic here is not right, in the scenario the API returns an empty array the Loading message will remain even when the HTTP request has been completed, this can give the user the false impression that the data is still being fetching, I recommend you to add a new boolean state `isLoading`  and set an initial value of true